pkgs/os-specific/linux/qtgreet/default.nix

```diff
@@ -0,0 +1,58 @@
+{
+  stdenv,
+  lib,
+
+  fetchFromGitLab,
+
+  wrapQtAppsHook,
+  meson,
+  pkg-config,
+  cmake,
+  qttools,
+  ninja,
+
+  qtbase,
+  json_c,
+  wlroots,
+  wayqt,
+  wayland,
+  pixman
+}:
+
+stdenv.mkDerivation (finalAtrrs: {
+  pname = "qtgreet";
+  version = "unstable-2022-01-24"; # Couldn't get 1.0.0 to build
+
+  src = fetchFromGitLab {
+    owner = "marcusbritanicus";
+    repo = finalAtrrs.pname;
```

Comment:
I think you can just use a recursive attrset here instead and just use `pname` instead of using `finalAttrs`

Reply:
I could, yeah. `finalAttrs` is better however. Recursive attrsets are kind of an anti-pattern.

Reply:
Did not know that, just saw it as the common way of doing things, the more you know